bot/cogs/birthday/birthday_commands.py

```diff
@@ -0,0 +1,118 @@
+from datetime import datetime, timezone, time
+import asyncio
+
+from nextcord import slash_command, Interaction, SlashOption, Permissions, Member, TextChannel
+from nextcord.ext import commands, tasks
+
+from bot.config import Config
+from bot.utils import messages
+from bot.utils.constants import TESTING_GUILD_ID, BUMPERS_GUILD_ID
+from db.helpers import birthday_helper
+
+class BirthdayCommands(commands.Cog):
+    def __init__(self, bot: commands.Bot):
+        self.bot = bot
+        self.post_birthdays.start()
+    
+    # Every day at 9AM Pacific Time
+    @tasks.loop(time=time(hour=17, minute=0, second=0, tzinfo=timezone.utc))
+    async def post_birthdays(self):
+        if not self.bot.is_ready():
+            return
+
+        birthdays = birthday_helper.get_todays_birthdays()
+        current_guild_id = None
+        birthday_channel_id = None
+        message = None
+        guild = None
+        for birthday in birthdays:
+            if birthday.guild_id != current_guild_id:
+                if message is not None and birthday_channel_id is not None:
+                    message = messages.get_special_birthday_fields(message)
+                    channel = self.bot.get_channel(birthday_channel_id)
+                    await channel.send(embed=message)
+                    message = None
+                guild = self.bot.get_guild(birthday.guild_id)
+                if guild is None:
+                    continue
+                current_guild_id = birthday.guild_id
+                birthday_channel_id = birthday_helper.get_birthday_channel_id(current_guild_id)
+                if birthday_channel_id is None:
+                    continue
+                message = messages.birthday_message()
+                member = guild.get_member(birthday.user_id)
+                message = messages.birthday_entry(message, birthday, member)
+            else:
+                member = guild.get_member(birthday.user_id)
+                message = messages.birthday_entry(message, birthday, member)
+        if message is not None and birthday_channel_id is not None: 
+            message = messages.get_special_birthday_fields(message)
+            channel = self.bot.get_channel(birthday_channel_id)
+            await channel.send(embed=message)
+
+
+    @post_birthdays.error
+    async def post_birthdays_error(self, e):
+        sentry_sdk.capture_exception(e)
+        await asyncio.sleep(60)
+        self.post_birthdays.restart()
+
+    @slash_command(name='birthday', guild_ids=[TESTING_GUILD_ID, BUMPERS_GUILD_ID],
+                   description='Birthday commands')
+    async def birthday(self, interaction: Interaction):
+        pass
+
+    @birthday.subcommand(name='add', description='Add a birthday')
+    async def birthday_add(self, interaction: Interaction,
+                            user: Member = SlashOption(name='user',
```

Comment:
It feels a bit odd for `user` to be a user-supplied argument. Do we want people to be able to setup birthdays for other users?

Reply:
I was thinking that would be useful from a mod perspective, but perhaps to keep things simpler we only allow individual users to control their own birthdays for now, and then could do some sort of birthday admin slash commands in the future if we actually find that they're necessary.

Reply:
Done, I removed user from both add and remove. 👍